### PR TITLE
testify: migrate test-infra to testify for analyze_test.go

### LINKED ISSUE #28564: migrate test-infra to testify for `analyze_test.go`

### LINKED ISSUE #30628: migrate test-infra to testify for `expression/analyze_test.go`

### PR DESCRIPTION
**Pull request by znhh6018**
### What problem does this PR solve?

Issue Number: close #28564 

close #30628

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by znhh6018** [comment]
@tisonkun ,some tests failed after I migrated,could I change the logic in the test to let it right. 
For example, the original logic is "require.NotNil(t, err)", but the test failed here , may I modify it as "require.NoError(t, err)"
and this is my first time to migrate test , there should be other bugs...
Thanks

**Comment by hawkingrei** [comment]
```
[2021-12-11T08:20:42.713Z] executor/analyze_test.go:434:2: ineffectual assignment to store (ineffassign)

[2021-12-11T08:20:42.713Z] 	store, clean := testkit.CreateMockStore(t)

```

Please fix this lint.

File: executor/analyze_test.go [hunk 1]
```diff
@@ -397,28 +428,31 @@ func checkHistogram(sc *stmtctx.StatementContext, hg *statistics.Histogram) (boo
...
 
-func (s *testFastAnalyze) TestFastAnalyze(c *C) {
-	c.Skip("Skip this unstable test(#25782) and bring it back before 2021-07-29.")
+func TestFastAnalyze(t *testing.T) {
+	t.Parallel()
+	t.Skip("Skip this unstable test(#25782) and bring it back before 2021-07-29.")
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
 	var cls testutils.Cluster
 	store, err := mockstore.NewMockStore(
```
Comment (hawkingrei) [inline executor/analyze_test.go:437]: You can remove this line.

Comment (znhh6018) [reply executor/analyze_test.go:433]: okay,you mean removing 434 and 435 ? I've removed just now.

File: executor/analyze_test.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018 PingCAP, Inc.
+// Copyright 2021 PingCAP, Inc.
```
Comment (hawkingrei) [inline executor/analyze_test.go:1]: It doesn't have to change.

File: executor/analyze_serial_test.go [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/collate"
+	"github.com/pingcap/tidb/util/israce"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFastAnalyze4GlobalStats(t *testing.T) {
+	if israce.RaceEnabled {
+		t.Skip("unstable, skip race test")
+	}
```
Comment (hawkingrei) [inline executor/analyze_serial_test.go:37]: It can be removed.

File: executor/analyze_test.go [hunk 4]
```diff
@@ -760,26 +660,36 @@ func (s *testFastAnalyze) TestFastAnalyzeRetryRowCount(c *C) {
...
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+
 	tk.MustExec("use test")
 	tk.MustExec("drop table if exists t")
 	tk.MustExec("create table t(a int primary key, b int, index index_b(b))")
 	tk.MustExec("set @@tidb_analyze_version = 1")
-	c.Assert(failpoint.Enable("github.com/pingcap/tidb/executor/buildStatsFromResult", `return(true)`), IsNil)
+	require.Nil(t, failpoint.Enable("github.com/pingcap/tidb/executor/buildStatsFromResult", `return(true)`))
```
Comment (hawkingrei) [inline executor/analyze_test.go:676]: require.NoError.

Please fix the other.

File: executor/analyze_test.go [hunk 5]
```diff
@@ -760,26 +660,36 @@ func (s *testFastAnalyze) TestFastAnalyzeRetryRowCount(c *C) {
...
 
-func (s *testSuite1) TestExtractTopN(c *C) {
+func TestExtractTopN(t *testing.T) {
+	t.Parallel()
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	dom, err := session.BootstrapSession(store)
+	require.NoError(t, err)
+	tk := testkit.NewTestKit(t, store)
 	if israce.RaceEnabled {
```
Comment (hawkingrei) [inline executor/analyze_test.go:690]: Skip can be removed.

**Comment by hawkingrei** [comment]
If fix the those, I think it is LGTM.

File: executor/analyze_test.go (hunk 2 above)
Comment (znhh6018) [reply executor/analyze_test.go:1]: done

**Comment by hawkingrei** [comment]
It has some reason for test to fail. Some test cases are unstable. the other may have some problem.

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
PTAL

/cc @tangenta @AilinKid

**Review by tangenta — APPROVED** [review]

**Pull request merged**